compiler/dyno/lib/parsing/parsing-queries.cpp

```diff
@@ -352,6 +353,81 @@ void setupModuleSearchPaths(Context* context,
   setModuleSearchPath(context, uSearchPath);
 }
 
+static void setChplEnv(Context* context, ChplEnvMap chplEnv) {
```

Comment:
I'm not quite sure I like the query structure here.

The main issue that I see is that the structure you have here will only correctly if `setupChplEnv` is run on before any other query on each revision. That means that if you incrementally recompile, you would need to run it each time. But, you do not have `testInteractive` doing that, for example. Otherwise, for example, the query system will try to recompute `getChplEnv` and run the body you have below that computes an empty map.

I do think that there are some situations in which setting a query is basically unavoidable but doing so always runs at cross-purposes with the query framework and incremental compilation more broadly. I do not think it is called for here for the entire Chapel Env map.

I think what we want to have happen is the following:
 1. We run `printchplenv` at all (which this PR does)
 2. We run `printchplenv` once per configuration even when doing incremental compilation; we can reuse the result of `printchplenv` across multiple incremental compiles and `printchplenv` itself takes a nontrivial time to run (on some platforms more than others)

Now, I think the design question here is - should dyno support different configurations (e.g. `CHPL_COMM=gasnet` and `CHPL_COMM=none`) in the same process? Or do we need/want a different process for any incremental compiler for each of these?

It's almost certainly too early to choose between these two but we have to pick one to start with for this PR.

If we want to have potentially many different configurations within a single process:
 * We would have `printchplenv` be modeled as a query that accepts an input environment mapping.
 * We would build up that input environment mapping including reading a `chplconfig` file (which would be an input query, probably)
 * We would assume that `printchplenv` will have the same result when given the same input (which is not quite true because e.g. an LLVM upgrade while a dyno process is running would mess it up, but I think we can expect a user to restart the compiler process in such a situation).

If we want to aim for just one configuration within each dyno process:
 * I recommend we just make `CHPL_HOME` an argument that needs to be provided when setting up the Context. We will want some way to opt out of passing a real one for the purpose of our C++ tests.
 * the Context can have a method to get the `ChplEnvMap` and it can run `printchplenv` if it hasn't been run yet; otherwise it would return the value saved from earlier
 * I don't think this should be a query since it shouldn't be recomputed on different incremental compiles.
 
I'd probably do the 2nd of these for now because it is simpler. We can move to the 1st if necessary later on.

Reply:
I'll opt for the second approach.